Makefile

```diff
@@ -1,147 +1,14 @@
-
-DOCKER_BUILD = docker build --no-cache \
-	       --build-arg user=$(USER) \
-	       --build-arg uid=$(shell id -u) \
-	       --build-arg guid=$(shell id -g) \
-	       --build-arg hostname=$(shell echo $$HOSTNAME) \
-	       -t $(SLAVE_IMAGE) \
-	       -f $(SLAVE_DIR)/Dockerfile.user \
-	       $(SLAVE_DIR) && \
-	       docker tag $(SLAVE_IMAGE):latest $(SLAVE_IMAGE):$(SLAVE_TAG)
-
-SONIC_BUILD_INSTRUCTION :=  make \
-                           -f slave.mk \
-                           PLATFORM=$(PLATFORM) \
-                           BUILD_NUMBER=$(BUILD_NUMBER) \
-                           ENABLE_DHCP_GRAPH_SERVICE=$(ENABLE_DHCP_GRAPH_SERVICE) \
-                           SHUTDOWN_BGP_ON_START=$(SHUTDOWN_BGP_ON_START) \
-                           SONIC_ENABLE_PFCWD_ON_START=$(ENABLE_PFCWD_ON_START) \
-                           ENABLE_SYNCD_RPC=$(ENABLE_SYNCD_RPC) \
-                           PASSWORD=$(PASSWORD) \
-                           USERNAME=$(USERNAME) \
-                           SONIC_BUILD_JOBS=$(SONIC_BUILD_JOBS) \
-                           KERNEL_PROCURE_METHOD=$(KERNEL_PROCURE_METHOD) \
-                           HTTP_PROXY=$(http_proxy) \
-                           HTTPS_PROXY=$(https_proxy) \
-                           SONIC_ENABLE_SYSTEM_TELEMETRY=$(ENABLE_SYSTEM_TELEMETRY)
-
-.PHONY: sonic-slave-build sonic-slave-bash init reset
-
-.DEFAULT_GOAL :=  all
+# SONiC make file
 
 %::
-	@docker inspect --type image $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG) &> /dev/null || \
-	    { echo Image $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG) not found. Building... ; \
-	    $(DOCKER_BASE_BUILD) ; }
-	@docker inspect --type image $(SLAVE_IMAGE):$(SLAVE_TAG) &> /dev/null || \
-	    { echo Image $(SLAVE_IMAGE):$(SLAVE_TAG) not found. Building... ; \
-	    $(DOCKER_BUILD) ; }
-	@$(INSMOD_OVERLAY)
-ifeq "$(KEEP_SLAVE_ON)" "yes"
-    ifdef SOURCE_FOLDER
-		@$(DOCKER_RUN) -v $(SOURCE_FOLDER):/var/$(USER)/src $(SLAVE_IMAGE):$(SLAVE_TAG) bash -c "$(SONIC_BUILD_INSTRUCTION) $@; /bin/bash"
-    else
-		@$(DOCKER_RUN) $(SLAVE_IMAGE):$(SLAVE_TAG) bash -c "$(SONIC_BUILD_INSTRUCTION) $@; /bin/bash"
-    endif
-else
-	@$(DOCKER_RUN) $(SLAVE_IMAGE):$(SLAVE_TAG) $(SONIC_BUILD_INSTRUCTION) $@
-endif
-
-sonic-slave-build :
-	$(DOCKER_BASE_BUILD)
-	$(DOCKER_BUILD)
-
-sonic-slave-bash :
-	@docker inspect --type image $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG) &> /dev/null || \
-	    { echo Image $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG) not found. Building... ; \
-	    $(DOCKER_BASE_BUILD) ; }
-	@docker inspect --type image $(SLAVE_IMAGE):$(SLAVE_TAG) &> /dev/null || \
-	    { echo Image $(SLAVE_IMAGE):$(SLAVE_TAG) not found. Building... ; \
-	    $(DOCKER_BUILD) ; }
-	@$(INSMOD_OVERLAY)
-	@$(DOCKER_RUN) -t $(SLAVE_IMAGE):$(SLAVE_TAG) bash
-
-showtag:
-	@echo $(SLAVE_IMAGE):$(SLAVE_TAG)
-	@echo $(SLAVE_BASE_IMAGE):$(SLAVE_BASE_TAG)
+	@echo "+++ --- Making $@ --- +++"
+	BLDENV=stretch make -f Makefile.work stretch
```

Comment:
can you add {if endif}, so that we can override the behavior, for example, NOSTRETCH=1 make will not make stretch target implicitly.

Reply:
Sure. Done. Thanks.